providers/openshift.md

```diff
@@ -32,9 +41,146 @@ master_hostname
+regardless of the hypervisor.
 
-openshift_use_manageiq is set to True to configure the Service Account needed by ManageIQ. there are more like variables like use_metrics to enable other options in OpenShift.
-A more detailed example will be found in inventory/byo/hosts.example.
-```ansible-playbook playbooks/byo/config.yml -i path/to/inventory/file```
+1. Create an addon directory for manageiq in minishift.  Something like:
+  
+  ```console
+  $ mkdir -p ./minifshift/addons/manageiq
+  ```
+  
+2. Create a `manageiq.addon` file with the following contents in that dir:
+  
+  ```bash
+  # Name: manageiq
+  # Description:  Basic Users, Roles, and Policies necessary to have ManageIQ/CFME connect to Openshift
+  
+  oc adm new-project management-infra --description="Management Infrastructure"
+  
+  oc create serviceaccount management-admin -n 'management-infra'
+  oc create serviceaccount inspector-admin -n 'management-infra'
+  
+  oc create -f management-infra-admin-cluster-role.json
+  oc create -f hawkular-metrics-admin-cluster-role.json
+  
+  oc policy add-role-to-user -n management-infra admin -z management-admin
+  oc policy add-role-to-user -n management-infra management-infra-admin -z management-admin
+  oc policy add-cluster-role-to-user cluster-reader system:serviceaccount:management-infra:management-admin
+  oc policy add-scc-to-user privileged system:serviceaccount:management-infra:management-admin
+  oc policy add-cluster-role-to-user system:image-puller system:serviceaccount:management-infra:inspector-admin
+  oc policy add-scc-to-user privileged system:serviceaccount:management-infra:inspector-admin
+  oc policy add-cluster-role-to-user self-provisioner system:serviceaccount:management-infra:management-admin
+  oc policy add-cluster-role-to-user hawkular-metrics-admin system:serviceaccount:management-infra:management-admin
+  ```
+  
+3. Create a management-infra-admin-cluster-role.json file with the following
+   contents in that dir:
+  
+  ```json
+  {
+    "apiVersion": "v1",
+    "kind":       "ClusterRole",
+    "metadata": {
+      "name": "management-infra-admin"
+    },
+    "rules": [
+      {"resources": ["pods/proxy"],"verbs":["*"]}
+    ]
+  }
+  ```
+  
+3. Create a hawkular-metrics-admin-cluster-role.json file with the following
+   contents in that dir:
+  
+  ```json
+  {
+    "apiVersion": "v1",
+    "kind":       "ClusterRole",
+    "metadata": {
+      "name": "hawkular-metrics-admin"
+    },
+    "rules": [
+      {
+        "apiGroups": [""],
+        "resources": ["hawkular-metrics","hawkular-alerts"],
+        "verbs":     ["*"]
+      }
+    ]
+  }
+  ```
+  
+4. Add and enable the manageiq addon to minishift:
+  
+  ```console
+  $ minishift addons install ./minishift/addons/manageiq
+  $ minishift addons enable manageiq
+  ```
+  
+5. Start minishift
+  
+  ```console
+  $ minishift start --vm-driver virtualbox  --openshift-version "v1.5.0-rc.0"
```

Comment:
Only did this for "copy pasta"-able instructions.  If you notice some of the other commands actually use the `oc` command that get's cached by minishift, so that was intentional.  Reduces the dependencies, but causes some weird workarounds.

Open to suggestions for a better way of handling this.